lib/internal/modules/esm/translators.js

```diff
@@ -333,16 +335,28 @@ translators.set('wasm', async function(url, source) {
     throw err;
   }
 
-  const imports =
+  const importNames = ArrayFrom(
+    new SafeSet(
       ArrayPrototypeMap(WebAssembly.Module.imports(compiled),
-                        ({ module }) => module);
+                        ({ module }) => module)));
+  const importNamespaces = await asyncESM.esmLoader.import(
+    importNames,
+    asyncESM.esmLoader.baseURL(url),
+    {}
+  );
+
   const exports =
     ArrayPrototypeMap(WebAssembly.Module.exports(compiled),
                       ({ name }) => name);
 
-  return createDynamicModule(imports, exports, url, (reflect) => {
-    const { exports } = new WebAssembly.Instance(compiled, reflect.imports);
+  const ret = new ModuleWrap(url, undefined, exports, function() {
+    const imports = ObjectCreate(null);
+    for (let i = 0; i < importNames.length; i++) {
+      imports[importNames[i]] = importNamespaces[i];
+    }
+    const { exports } = new WebAssembly.Instance(compiled, imports);
     for (const expt of ObjectKeys(exports))
```

Comment:
shouldn't this use a for loop too, because of Symbol.iterator?

Reply:
probably, but wasn't doing touch up there, could change if desired.